js-library/src/request-verifiable-presentation.ts

```diff
@@ -196,28 +210,31 @@ export const requestVerifiablePresentation = ({
     );
   };
   const nextFlowId = createFlowId();
+  currentFlows.set(nextFlowId, "initialized");
   const handleCurrentFlow = handleFlowFactory(nextFlowId);
   window.addEventListener("message", handleCurrentFlow);
   const url = new URL(identityProvider);
   url.pathname = "vc-flow/";
   // As defined in the spec: https://github.com/dfinity/internet-identity/blob/main/docs/vc-spec.md#1-load-ii-in-a-new-window
   const iiWindow = window.open(url, "idpWindow", windowOpenerFeatures);
-  // Check if the _idpWindow is closed by user.
-  const checkInterruption = (flowId: FlowId): void => {
-    // The _idpWindow is opened and not yet closed by the client
-    if (iiWindow) {
-      if (iiWindow.closed && currentFlows.has(flowId)) {
+  if (iiWindow !== null) {
+    iiWindows.set(nextFlowId, iiWindow);
+    // Check if the _idpWindow is closed by user.
+    const checkInterruption = (flowId: FlowId): void => {
+      // The _idpWindow is opened and not yet closed by the client
+      if (
+        iiWindow.closed &&
```

Comment:
Would it make sense to rename `iiWindow` to `idpWindow`?

Reply:
Yes, a lot of sense. I kept it as Identity Provider in all comments as well.